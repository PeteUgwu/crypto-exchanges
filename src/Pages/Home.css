.main-container {
  text-align: center;
}

.search-field {
  padding: 3rem 0;
  background-color: #fe5395;
  margin: 0;
}

.search-field input {
  border-radius: 20px;
  width: 15rem;
  height: 1.5rem;
  padding: 1rem;
  background-color: #ec4c8b;
  color: #fff;
  border-style: none;
  border-color: #df4783;
  text-align: center;
}

input::placeholder {
  font-style: italic;
  color: #fff;
}

.exchange-container {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 0.2rem;
  padding: 1rem 0.5rem;
  justify-content: space-between;
}
