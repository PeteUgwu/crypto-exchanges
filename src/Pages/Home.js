import React from 'react'
import { useSelector } from 'react-redux';
import ExchangeDetails from '../Components/ExchangeDetails';

const Home = () => {
    const exchanges = useSelector((state) => state.Exchanges);
    const showDetails = (id) => {
       

    }
  return (
    <div>
        <div>
            <input type="text" value="text" placeholder="Search An Exchange" />
        </div>
        <div>
            {exchanges.map((exchange) => (
                <ExchangeDetails 
              key={exchange.id}
              exchange={exchange}
              showDetails={showDetails} 
            />
            ))}
        </div>
    </div>
  )
}

export default Home;