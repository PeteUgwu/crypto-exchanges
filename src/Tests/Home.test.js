import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../Redux/ConfigureStore';
import Home from '../Pages/Home';

it('snapshots', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Home />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
