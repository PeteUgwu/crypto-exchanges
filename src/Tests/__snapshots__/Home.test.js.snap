// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<div
  className="main-container"
>
  <div
    className="search-field"
  >
    <input
      onChange={[Function]}
      placeholder="Search An Exchange"
      type="text"
    />
  </div>
  <div
    className="exchange-container"
  />
</div>
`;
